File: 백준/Silver/11399. ATM/ATM.rs
use std::io;

fn main() {
    let mut input_a = String::new();
    io::stdin().read_line(&mut input_a).unwrap();
    let n: Vec<i32> = input_a
        .split_whitespace()
        .map(|x| -> i32 { x.parse().unwrap() })
        .collect();
    let mut input_b = String::new();
    io::stdin().read_line(&mut input_b).unwrap();
    let mut numbers: Vec<i32> = input_b
        .split_whitespace()
        .map(|x| -> i32 { x.parse().unwrap() })
        .collect();
    numbers.sort();
    let mut time = 0;
    let mut tmp = 0;

    for i in 0..n[0] {
        tmp += numbers[i as usize];
        time += tmp;
    }
    println!("{}", time);
}

File: 백준/Silver/11399. ATM/ATM.py
import sys

reader = sys.stdin.readline
writer= sys.stdout
n= int(reader())
nums = list(map(int, reader().split()))

nums.sort()

time =0
tmp = 0

for i in range(n):
    tmp+=nums[i]
    time+=tmp
writer.write(str(time) + '\n')
